constraint-solver/src/grouped_expression.rs

```diff
@@ -231,7 +232,7 @@ impl<T: RuntimeConstant, V: Ord + Clone + Eq> GroupedExpression<T, V> {
         (&self.quadratic, self.linear.iter(), &self.constant)
     }
 
-    /// Computes the degree of a GroupedExpression (as it is contsructed) in the unknown variables.
```

Comment:
I don't think it matters how it is constructed?

Reply:
It does: `(a * b) - 3 * (a * b)  + 2 * (a * b)` has degree zero, but this function will return two.

Reply:
I see, good point! In that case, I think we can be more precise, I added this comment: f9e977eec4b8b1cbc6240650b76ca21424aed5a9

Reply:
Maybe something like `a * (b + 1) - a * b - a`